lib/matplotlib/__init__.py

```diff
@@ -215,7 +217,78 @@ def _is_writable_dir(p):
     """
     return os.access(p, os.W_OK) and os.path.isdir(p)
 
+_verbose_msg = """--verbose-LEVEL and the Verbose class are deprecated.
+                  This functionality is now provided by the standard
+                  python logging library.  To get more (or less) logging
+                  output:
+                  `import logging`
+                  `logger=logging.getLogger('matplotlib')`
+                  `logger.set_level(logging.INFO)`
+                  """
 
+
+def _set_logger_verbose_level(level_str='silent', file_str='sys.stdout'):
+    """
+    Use a --verbose-LEVEL level to set the logging level:
+
+    """
+    levelmap = {'silent': logging.WARNING, 'helpful': logging.INFO,
+                'debug': logging.DEBUG, 'debug-annoying': logging.DEBUG,
+                'info': logging.INFO, 'warning': logging.warning}
+    # Check that current state of logger isn't already more verbose
+    # than the requested level.  If its more verbose, then leave more
+    # verbose.
+    if level_str in levelmap:
+        log = logging.getLogger(__name__)
+        newlev = levelmap[level_str]
+        oldlev = log.getEffectiveLevel()
+        if newlev < oldlev:
+            log.setLevel(newlev)
+            std = {
+                'sys.stdout': sys.stdout,
+                'sys.stderr': sys.stderr,
+            }
+            if file_str in std:
+                fileo = std[file_str]
+            else:
+                fileo = sys.stdout
+                try:
+                    fileo = open(file_str, 'w')
+                    # if this fails, we will just write to stdout
+                except IOError:
+                    warning.warn('could not open log file "{0}"'
+                                'for writing.\nCheck your '
+                                'matplotlibrc'.format(file_str))
+            console = logging.StreamHandler(fileo)
+            console.setLevel(newlev)
+            log.addHandler(console)
+
+
+def _parse_commandline():
+    """
+    Check for --verbose-LEVEL type command line arguments and
+    set logging level appropriately.
+    """
+    log = logging.getLogger(__name__)
+
+    for arg in sys.argv[1:]:
+        # cast to str because we are using unicode_literals,
+        # and argv is always str
```

Comment:
I don't think you need the cast?  On Py2 the str will just happily check for startswith against unicode; on Py3 everything is str.
If you do need the cast I'd just write `for arg in map(str, sys.argv[1:])`

Reply:
This might a 2.6 hold over?

This works as expected on as far back as 2.7.3 (the earliest version conda had easily available).

Reply:
Works with 2.6.8 as well, I do wonder how far back this goes...

Reply:
I'm going to leave this in as I just copied from old code and don't understand the cast.  Feel free to tell me the proper way to do it.